shiksha-api/app-service/app/routers/chat_mcp.py

```
@@ -34,15 +42,22 @@ async def chat_general(
 
-            response_content = await GENERAL_CHAT_SERVICE_INSTANCE([
+            content = io.StringIO()
+            async for event_raw in general_chat_svc([
                 ConversationMessage(role=MessageRole.USER, message=message)
-            ], user_id=user_id)
+            ], user_id=user_id):
+                event = json.loads(event_raw)
+                if event["type"] == "content":
+                    content.write(event["delta"])
+                elif event["type"] == "error":
+                    raise ToolError(event["data"])
```

Ruff (0.15.21)

[warning] Abstract `raise` to an inner function

(TRY301)